### PR TITLE
Extended template language to allow some nesting in if/else and fix sound park search using lidarr.

### LINKED ISSUE #6780: Soundpark path change
The path music for sound park no longer works (https://sound-park.world/music?q=National&num=50)
https://github.com/Jackett/Jackett/blob/8a6d3c6c2e898d7db3cc3955591c65f64ad3197f/src/Jackett.Common/Definitions/soundpark.yml#L41 
needs to be always search (https://sound-park.world/search?q=National&num=50)

This happens when the Torznab search is made by artist and album. 
I also think the album should be added to the keywords if exists, not just the artist.
https://github.com/Jackett/Jackett/blob/8a6d3c6c2e898d7db3cc3955591c65f64ad3197f/src/Jackett.Common/Definitions/soundpark.yml#L43

### PR DESCRIPTION
**Pull request by lightglitch**
Fixes #6780

### CODE REVIEW

File: src/Jackett.Common/Definitions/soundpark.yml [hunk 1]
```diff
@@ -38,9 +38,9 @@
 
   search:
     paths:
-      - path: "{{if .Keywords}}search{{else}}music{{end}}"
+      - path: "search"
     inputs:
-      q: "{{if .Query.Artist}}{{ .Query.Artist }}{{else}}{{ .Keywords }}{{end}}"
+      q: "{{if or (.Query.Album) (.Query.Artist) }}{{ or (.Query.Album) (.Query.Artist) }}{{else}}{{ .Keywords }}{{end}}"
```
Comment (garfield69) [inline src/Jackett.Common/Definitions/soundpark.yml]: we still need to support path music for when lidarr users add the indexer for the first time and/or do a TEST (which as far as I believe lidarr just sends a keyword-less query to jackett), as well as users that are using the jackett dashboard to do a search without keywords.
does the and/or support allow for coding a path that is the equivalent of this non-literal example:
-path "{{if .Keywords or .Query.Artist or .Query.Album}}search{{else}}music{{end}}" ?

**Comment by lightglitch** [comment]
The implemented syntax for "and" a "or" it's was very simple, I just added support for one more argument to support what you want.

It's not a normal "or" or "and" it's limited to 2-3 arguments and returns the value of one of the arguments, doesn't return a boolean. And the syntax is very strict because of the replace.

**Comment by garfield69** [comment]
can you test your latest change against using the jackett dashboard to search soundpark with keywords?
in my dev box with your changes applies, if I search for `Nation` the following url is generated `https://sound-park.world/music?q=Nation&num=50`

a keyword-less search on the dashboard generates `https://sound-park.world/music?q=&num=50` so that is working ok.
and using lidarr to searching for the album Super from the Pet Shop Boys generates `https://sound-park.world/search?q=Super&num=50` so this looks good to me too.

**Comment by lightglitch** [comment]
I did tested the changes before updating the pull request and the behavior is the same as yours.
When using lidarr I decided to only send the album or artist because if we send both soundpark may not return results.

**Comment by lightglitch** [comment]
@garfield69 need anything else?

**Comment by garfield69** [comment]
well yes. a fix for using the jackett dashboard to search soundpark with keywords.
A search for **Nation** returns the music page not the search page with National results.

at the moment your code works only for lidarr, and for dashboard search with no keywords.
if this is not possible using your if/else extension then I'll have to create two indexer, one exclusively for lidarr the other for dashboard.

**Comment by lightglitch** [comment]
@garfield69 Sorry, I didn't understand there was a bug from your previous comment. It's fixed now.

**Comment by garfield69** [comment]
Brilliant! Thank you.

**Pull request merged**